### PR TITLE
tests: add test for adding invalid cores devices to caches

### PR DESCRIPTION
**Pull request by DocentSzachista**
Signed-off-by: Damian Raczkowski <damian.raczkowski@intel.com>

### CODE REVIEW

File: test/functional/tests/ci/test_basic.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    pass criteria:
+      - Doesn't start additional cache on the same device
+      - Doesn't start cache on another device with the same id
+    """
+    with TestRun.step("Set up device"):
+        cache_device_1 = TestRun.disks["cache_1"]
+        cache_device_1.create_partitions([Size(2000, Unit.MebiByte)])
+        cache_device_2 = TestRun.disks["cache_2"]
+        cache_device_2.create_partitions([Size(2000, Unit.MebiByte)])
+    with TestRun.step("Start cache with ID=1 on cache device"):
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:30]: You do not need to specify that id on that step.

**Review by jfckm — APPROVED** [review]

File: test/functional/tests/ci/test_basic.py [hunk 2]
```diff
@@ -32,3 +35,38 @@ def test_cas_version():
...
+    pass criteria:
+      - Does start cache on cache device
+      - Doesn't start cache on the same device with another id
+      - Doesn't start cache on another partition with the same id
+    """
+    with TestRun.step("Set up device"):
+        cache_dev = TestRun.disks["cache_1"]
+        cache_dev.create_partitions([Size(2000, Unit.MebiByte), Size(2000, Unit.MebiByte)])
+        partition_1 = cache_dev.partitions[0]
+        partition_2 = cache_dev.partitions[1]
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:55]: We pretty often align to convention:
```
cache_dev = TestRun.disks["cache_1"]
cache_dev.create_partitions([Size(2000, Unit.MebiByte)])
cache_dev = cache_dev.partitions[0]
```
using shadowing variables. So cache_dev is most often name for cache device as partition. Consider in context of test maybe convention of `cache_dev_used` or `cache_dev_occupied` and `cache_dev_free`.

File: test/functional/tests/ci/test_basic.py [hunk 3]
```diff
@@ -32,3 +35,38 @@ def test_cas_version():
...
+
+    with TestRun.step("Start cache on the same device but with another ID"):
+        output = TestRun.executor.run_expect_fail(
+            cli.start_cmd(partition_1.path, cache_id="2", force=True)
+        )
+        cli_messages.check_stderr_msg(output, cli_messages.start_cache_on_already_used_dev)
+
+    with TestRun.step("Start cache with ID=1 on another cache device"):
+        output = TestRun.executor.run_expect_fail(
+            cli.start_cmd(partition_2.path, cache_id="1", force=True)
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:70]: Consider additional variables for `cache_id`

**Comment by jfckm** [comment]
@CAS-Linux-Jenkins ok to test

**Review by pdebski21 — APPROVED** [review]
Looks fine. Good job (:

**Review by jfckm — APPROVED** [review]

File: test/functional/tests/ci/test_basic.py [hunk 4]
```diff
@@ -32,3 +40,40 @@ def test_cas_version():
         if splitted_version[1] != file_cas_version:
             TestRun.LOGGER.error(f"""Version of {splitted_version[0]} from cmd doesn't match
              with file. Expected: {file_cas_version} Actual: {splitted_version[1]}""")
+
+
+@pytest.mark.require_disk("cache_1", DiskTypeSet([DiskType.nand, DiskType.optane]))
```
Comment (karolinavelkaja) [inline test/functional/tests/ci/test_basic.py:45]: is this test for CI? If yes please add proper pytest mark. If not just answear and I will marge this PR:)

Comment (karolinavelkaja) [reply test/functional/tests/ci/test_basic.py:45]: If I've checked correctly, looks like this should be part of new CI scpe, so please add:
@pytest.mark.CI

**Review by karolinavelkaja — CHANGES_REQUESTED** [review]
If I've checked correctly, looks like this should be part of new CI scope, so please add:
@pytest.mark.CI

**Comment by DocentSzachista** [comment]
@karolinavelkaja Added :)

**Review by jfckm — APPROVED** [review]

**Review by pdebski21 — APPROVED** [review]

**Review by karolinavelkaja — APPROVED** [review]

**Pull request merged**